src/rttr/library.cpp

```diff
@@ -29,6 +29,9 @@
 #include "rttr/detail/library/library_p.h"
 
 #include <map>
+#include <mutex>
+
+static std::mutex g_library_mutex;
```

Comment:
wouldn't it be better to put this variable in the `library_manager` since it is already a singleton and physically belongs to it (why a global?, ok it is a `static` meaning only in this translation unit when I understand correct) and thread-safe initialized because of the magic `static`s `static library_manager obj;`